realsense2_camera/test/live_camera/test_camera_all_profile_tests.py

```diff
@@ -0,0 +1,291 @@
+    'camera_name': 'D415',
+    'device_type': 'D415',
+    }
+test_params_all_profiles_d435 = {
+    'camera_name': 'D435',
+    'device_type': 'D435',
+    }
+'''
+This test was implemented as a template to set the parameters and run the test.
+This directory is not added to the CMakeLists.txt so as to avoid the colcon failure in the
+machines that don't have the D455 connected.
+1. Only a subset of parameter types are implemented in py_rs_utils, it has to be extended for others
+2. After setting the param, rclpy.spin_once may be needed.Test passes even without this though.
+'''
+@pytest.mark.parametrize("launch_descr_with_parameters", [test_params_all_profiles_d435, test_params_all_profiles_d415,test_params_all_profiles_d455],indirect=True)
+@pytest.mark.launch(fixture=launch_descr_with_parameters)
+class TestLiveCamera_Change_Resolution(pytest_rs_utils.RsTestBaseClass):
+    def test_LiveCamera_Change_Resolution(self,launch_descr_with_parameters):
+        skipped_tests = []
+        params = launch_descr_with_parameters[1]
+        themes = [
+        {'topic':'/'+params['camera_name']+'/color/image_raw',
+         'msg_type':msg_Image,
+         'expected_data_chunks':1,
+         #'data':data
+        }
+        ]
+        depth_themes = [
+        {'topic':'/'+params['camera_name']+'/depth/image_rect_raw',
+         'msg_type':msg_Image,
+         'expected_data_chunks':1,
+         #'data':data
+        }
+        ]  
+        infra_themes = [
+        {'topic':'/'+params['camera_name']+'/infra/image_rect_raw',
+         'msg_type':msg_Image,
+         'expected_data_chunks':1,
+         #'data':data
+        }
+        ]        
+        infra1_themes = [
+        {'topic':'/'+params['camera_name']+'/infra1/image_rect_raw',
+         'msg_type':msg_Image,
+         'expected_data_chunks':1,
+         #'data':data
+        }
+        ]   
+        infra2_themes = [
+        {'topic':'/'+params['camera_name']+'/infra2/image_rect_raw',
+         'msg_type':msg_Image,
+         'expected_data_chunks':1,
+         #'data':data
+        }
+        ]   
+        try:
+            ''' 
+            initialize, run and check the data 
+            '''
+            serial_no = None
+            if 'serial_no' in params:
+                serial_no = params['serial_no']
+            self.init_test("RsTest"+params['camera_name'])
+            cap = pytest_live_camera_utils.get_camera_capabilities(params['device_type'], serial_no)
+            if cap == None:
+                print("Device not found? : " + params['device_type'])
+                return
+            self.create_param_ifs(params['camera_name'] + '/' + params['camera_name'])
+            for profile in cap["color_profile"]:
+                if profile[0] == 'Color':
+                    print("Testing :"+ profile[0] + " " + profile[1] + " " + profile[2])
+                    if "BGRA8" == profile[2]:
+                        print("Skipping " +profile[2])
```

Comment:
I added support for these type of formats in our node.
Still in review.. maybe when we merge my PR you will remove this "skipping"